packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -185,6 +185,13 @@ export interface CommonAutoScalingGroupProps {
    * @default - Uses the block device mapping of the AMI
    */
   readonly blockDevices?: BlockDevice[];
+
+  /**
+   * The maximum amount of time that an instance can be in service.
+   *
+   * @default none
+   */
+  readonly maxInstanceLifetime?: Duration;
```

Comment:
Please add more documentation, specifically the allowed values and a link to ec2 [docs](https://docs.aws.amazon.com/autoscaling/ec2/userguide/asg-max-instance-lifetime.html)